client/components/gravatar/index.jsx

```diff
@@ -1,37 +1,46 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component } from 'react';
 import url from 'url';
 import qs from 'querystring';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import safeImageURL from 'lib/safe-image-url';
+import {
+	getUserTempGravatar
+} from 'state/current-user/gravatar-status/selectors';
 
-module.exports = React.createClass( {
-	displayName: 'Gravatar',
+export class Gravatar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			failedToLoad: false
+		};
+	}
 
-	propTypes: {
+	static displayName = 'Gravatar';
+
+	static propTypes = {
 		user: React.PropTypes.object,
 		size: React.PropTypes.number,
-		imgSize: React.PropTypes.number
-	},
+		imgSize: React.PropTypes.number,
+		// connected props:
+		tempImage: React.PropTypes.oneOfType( [
```

Comment:
Why can be the `tempImage` both a string and a bool? We could maybe add some inline comment to clarify further. :)